crates/bevy_ecs/src/query/iter.rs

```diff
@@ -432,17 +415,20 @@ struct QueryIterationCursor<'w, 's, Q: WorldQuery, F: WorldQuery> {
     phantom: PhantomData<Q>,
 }
 
-impl<'w, 's, Q: WorldQuery, F: WorldQuery> Clone for QueryIterationCursor<'w, 's, Q, F>
-where
-    QueryFetch<'w, Q>: Clone,
-    QueryFetch<'w, F>: Clone,
-{
-    fn clone(&self) -> Self {
+impl<'w, 's, Q: WorldQuery, F: WorldQuery> QueryIterationCursor<'w, 's, Q, F> {
+    /// This function is safe to call if `(Q, F): ReadOnlyWorldQuery` holds.
```

Comment:
Ditto here; do we want to split out another method with that trait bound?

Reply:
I don't think it really matters, there was literaly one call site afaict and it doesnt know that `(Q, F): ReadOnlyWorldQuery` holds